zerver/lib/import_realm.py

```diff
@@ -1979,6 +1981,15 @@ def update_message_foreign_keys(import_dir: Path, sort_by_date: bool) -> None:
             new_id=new_id,
         )
 
+    # Save ID_MAP to disk so parallel workers can load it
+    # Convert integer keys to strings for JSON compatibility
+    id_map_path = os.path.join(import_dir, "id_map.json")
+    serializable_id_map = {
+        table: {str(k): v for k, v in mapping.items()} for table, mapping in ID_MAP.items()
+    }
+    with open(id_map_path, "wb") as f:
+        f.write(orjson.dumps(serializable_id_map))
+
```

Comment:
There is no need to serialize this to disk; see the `initializer` and `initargs` arguments to `run_parallel`, which let you just pass the Python data structure directly into the process.